src/aap_eda/analytics/utils.py

```diff
@@ -23,3 +35,72 @@ def datetime_hook(dt: dict) -> dict:
         except TypeError:
             new_dt[key] = value
     return new_dt
+
+
+def collect_controllers_info() -> dict:
+    aap_credentia_type = models.CredentialType.objects.get(
+        name=enums.DefaultCredentialType.AAP
+    )
+    credentials = models.EdaCredential.objects.filter(
+        credential_type=aap_credentia_type
+    )
+    info = {}
+    for credential in credentials:
+        controller_info = {}
+        inputs = yaml.safe_load(credential.inputs.get_secret_value())
+        host = inputs["host"]
+        url = f"{host}/api/v2/ping/"
+        verify = str_to_bool(inputs.get("verify_ssl", ""))
+        token = inputs.get("oauth_token")
+
+        controller_info["credential_id"] = credential.id
+        controller_info["inputs"] = inputs
+        if token:
+            headers = {"Authorization": f"Bearer {token}"}
+            logger.info("Use Bearer token to ping the controller.")
+        else:
+            user_pass = f"{inputs.get('username')}:{inputs.get('password')}"
+            auth_value = (
+                f"Basic {base64.b64encode(user_pass.encode()).decode()}"
+            )
+            headers = {"Authorization": f"{auth_value}"}
+            logger.info("Use Basic authentication to ping the controller.")
+
+        try:
+            resp = requests.get(url, headers=headers, verify=verify)
+            resp_json = resp.json()
+            controller_info["install_uuid"] = resp_json["install_uuid"]
+
+            info[host] = controller_info
+        except requests.exceptions.RequestException as e:
+            logger.warning(
+                f"Credential {credential.name} failed in authorization: {e}"
+            )
+
+    return info
+
+
+def extract_job_details(
+    url: str,
+) -> Tuple[Optional[str], Optional[str], Optional[str]]:
+    controllers = collect_controllers_info()
```

Comment:
For each job (url) you call collect_controllers_info() which does not change. We should only call it once per analytics task.